### ISSUE TITLE
Fail to install Horovod with MXNet 1.5.1 and nightly

### ISSUE THREAD

**Issue by apeforest** [apache/incubator-mxnet#16193]
The symbol _ZN5mxnet12on_enter_apiEPKc is missing from mxnet-cu100mkl nightly and 1.5.1 package. Without it, horovod cannot run.

This symbol is available in mxnet-cu100 1.5.0 package.

**Comment by apeforest** [comment]
@szha @TaoLv mxnet 1.5.1 pip package has the same issue. Horovod cannot be installed with MXNet 1.5.1. 

Also when I do `nm -D lib/python3.6/site-packages/mxnet/libmxnet.so` I got nm: File format not recognized. Same command works with 1.5.0 so file. Is there anything changed in the release pipeline? Thanks!

**Comment by leezu** [comment]
`_ZN5mxnet10CopyFromToERKNS_7NDArrayEPS1_i` is missing as well. (Adding here for SEO)

**Comment by szha** [comment]
@apeforest @leezu I think the problem is that starting recently we used upx for compressing the binary.

**Comment by szha** [comment]
I'm releasing a post0 version of 1.5.1 to remove the upx compression to mitigate the impact. We still need to figure out a way to link horovod to the compressed libmxnet

**Comment by szha** [comment]
A summary of the actions being taken:
- A post-release version has been released for all variants for 1.5.1.
- We requesting an increase to the package size limit so that we could remove the compression for nightly. https://github.com/pypa/warehouse/issues/6763
- @reminisce is investigating how we can compile the operators into a separate binary on which compression can again be applied.

**Comment by leezu** [comment]
We should agree on some solution prior to the next release.

**Comment by ptrendx** [comment]
Hi @leezu, you added the R1.6.0 label here, but I believe this is not a blocker for the release (and it seems to be kind of resolved now anyway) - is that true?

**Comment by szha** [comment]
@ptrendx the new nightly on s3 should work with Horovod now as it doesn't include binary compression with upx. Since the limit is already raised, we should be able to release 1.6.0 correctly too.

@apeforest please verify and reopen if necessary.

**Comment by leezu** [comment]
@ptrendx I think it is a blocker because we can't release eto pypi with this bug open. But yes, it's resolved.

**Comment by apeforest** [comment]
@szha Yes, I can verify that Horovod can be built successfully using the nightly pip package from https://apache-mxnet.s3-us-west-2.amazonaws.com/dist/2019-12-07/dist/mxnet-1.6.0b20191207-py2.py3-none-manylinux1_x86_64.whl. Can we have a command to install mxnet nightly without having to specify a particular date. I can then add test to horovod nightly test using the mxnet nightly build. Thanks!

**Comment by szha** [comment]
sure. I plan to first simplify the path as it's not useful to have two `dist` in the path. Then we can probably make a mxnet-nightly-latest.whl file which gets overwritten with the new releases.

**Comment by ChaiBapchya** [comment]
Bumped into the same error for
```
mxnet-cu101==1.6.0
```

Repro steps
```
    virtualenv -p python3 mx16cu101
    source mx16cu101/bin/activate
    pip install mxnet-cu101==1.6.0
    pip install gluoncv
    pip install horovod
    cd incubator-mxnet/
    python example/distributed_training-horovod/resnet50_imagenet.py
```

**Comment by ChaiBapchya** [comment]
Can we re-open the issue @leezu @apeforest @szha

**Comment by ChaiBapchya** [comment]
Nevermind, realized that nightly builds arent published to PyPi anymore. But that leaves many people who consume binaries via PyPi to lose out on leveraging horovod with existing versions.

**Comment by leezu** [comment]
If it is broken in the stable version on pypi, it should be fixed. Can you open a new issue about that if it's broken?

**Comment by ChaiBapchya** [comment]
It also fails for nightly
I bumped into the same issue for wheels found in repo.mxnet.io
```
pip install https://repo.mxnet.io/dist/python/cu102/mxnet_cu102-1.6.0-py2.py3-none-manylinux1_x86_64.whl
pip install https://repo.mxnet.io/dist/python/cu102/mxnet_cu102-1.7.0b20200719-py2.py3-none-manylinux2014_x86_64.whl
pip install https://repo.mxnet.io/dist/python/cu102mkl/mxnet_cu102mkl-1.6.0b20200215-py2.py3-none-manylinux1_x86_64.whl
```
